app/routes/xiview_xi2.py

```
@@ -0,0 +1,304 @@
+    query = "SELECT intensity, mz FROM spectrum WHERE id = $1 AND spectra_data_id = $2 AND upload_id = $3"
+    data = await execute_query(query, [id, int(sd_ref), int(upload_id)], fetch_one=True)
+    # Create a new dictionary to store the unpacked values
+    unpacked_data = {
+        "intensity": struct.unpack('%sd' % (len(data['intensity']) // 8), data['intensity']),
+        "mz": struct.unpack('%sd' % (len(data['mz']) // 8), data['mz'])
+    }
+    return Response(orjson.dumps(unpacked_data), media_type='application/json')
+
+
+@xiview_xi2_data_router.post('/get_annotated_peaklist', tags=["xiVIEW"])
+async def get_annotated_peaklist(request: Request, id: str, sd_ref: str, upload_id: str):
```

Ruff (0.15.14)

[error] Function argument `id` is shadowing a Python builtin

(A002)